src/main/java/org/jabref/gui/entryeditor/FileAnnotationTab.java

```diff
@@ -267,52 +287,91 @@ private JPanel setUpButtons() {
+        sj.add(Localization.lang("Marking") + ": " + markedTxtArea.getText());
 
         new ClipBoardManager().setClipboardContents(sj.toString());
     }
 
     private void reloadAnnotations() {
         isInitialized = false;
         Arrays.stream(this.getComponents()).forEach(this::remove);
+        fileAnnotationCache.remove(parent.getEntry());
         initializeTab(this);
         this.repaint();
     }
 
 
     /**
-     * Fills the highlight and annotation texts and enables/disables the highlight area if there is no highlighted text
+     * Fills the TextAreas of the content and the highlighted or underlined text with the corresponding text and also
+     * changes the label accordingly.
      *
-     * @param annotation either a text annotation or a highlighting from a pdf
+     * @param annotation either a text annotation or a marking from a PDF
      */
-    private void updateContentAndHighlightTextAreas(final FileAnnotation annotation) {
+    private void updateContentAndMarkedTextAreas(final FileAnnotation annotation) {
+        updateMarkingType(annotation);
 
         if (annotation.hasLinkedAnnotation()) {
             // isPresent() of the optional is already checked in annotation.hasLinkedAnnotation()
-            if (!annotation.getLinkedFileAnnotation().getContent().isEmpty()) {
-                contentTxtArea.setText(annotation.getLinkedFileAnnotation().getContent());
+            if (!delocalize(annotation.getLinkedFileAnnotation().getContent()).isEmpty()) {
+                contentTxtArea.setText(delocalize(annotation.getLinkedFileAnnotation().getContent()));
                 contentTxtArea.setEnabled(true);
             } else {
                 contentTxtArea.setText("N/A");
                 contentTxtArea.setEnabled(false);
             }
 
-            if (annotation.getContent().isEmpty()) {
-                highlightTxtArea.setEnabled(false);
-                highlightTxtArea.setText("JabRef: The highlighted area does not contain any legible text!");
+            if (delocalize(annotation.getContent()).isEmpty()) {
+                markedTxtArea.setEnabled(false);
+                markedTxtArea.setText(Localization.lang("The marked area does not contain any legible text!"));
             } else {
-                highlightTxtArea.setEnabled(true);
-                highlightTxtArea.setText(annotation.getContent());
+                markedTxtArea.setEnabled(true);
+                markedTxtArea.setText(delocalize(annotation.getContent()));
             }
-
         } else {
-            contentTxtArea.setText(annotation.getContent());
-            highlightTxtArea.setText("N/A");
-            highlightTxtArea.setEnabled(false);
+            contentTxtArea.setText(delocalize(annotation.getContent()));
+            contentTxtArea.setEnabled(true);
+            markedTxtArea.setText("N/A");
+            markedTxtArea.setEnabled(false);
+        }
+    }
+
+    private String delocalize(String content) {
```

Comment:
Because the `DefaultListModel<FileAnnotation> listModel;` uses the `FileAnnotation.toString()` method directly to render the `JList<FileAnnotation> annotationList;`. There we need the localized name of the `AnnotationType`, whereas in the `private final JTextArea markedTxtArea;` this would be a duplication, so I stripped it.

I'm not proficient in GUI programming, and this was the easiest solution I found.